docs/cub/developer_overview.rst

```diff
@@ -386,282 +386,264 @@ Device-scope
+and passes it to the ``ChainedPolicy::Invoke`` function:
 
 .. code-block:: c++
 
-    template <typename InputIteratorT,
-              // ...
-              typename SelectedPolicy>
-    struct DispatchReduce : SelectedPolicy
-    {
-
-      //
-      CUB_RUNTIME_FUNCTION __forceinline__ static
-      cudaError_t Dispatch(
-          void *d_temp_storage, size_t &temp_storage_bytes,
-          InputIteratorT d_in,
-          /* ... */)
-      {
-        typedef typename DispatchSegmentedReduce::MaxPolicy MaxPolicyT;
-
-        if (num_segments <= 0) return cudaSuccess;
-
-        cudaError error = cudaSuccess;
-        do {
-          // Get PTX version
-          int ptx_version = 0;
-          error = CubDebug(PtxVersion(ptx_version));
-          if (cudaSuccess != error)
-          {
-            break;
-          }
-
-          // Create dispatch functor
-          DispatchSegmentedReduce dispatch(
-              d_temp_storage, temp_storage_bytes, d_in, /* ... */);
-
-          // Dispatch to chained policy
-          MaxPolicyT::Invoke(ptx_version, dispatch);
-        } while (0);
-
-        return error;
+    template <..., // algorithm specific compile-time parameters
+              typename SelectedPolicy> // also called: PolicyHub
+    struct DispatchAlgorithm : SelectedPolicy { // TODO(bgruber): I see no need for inheritance, can we remove it?
+      CUB_RUNTIME_FUNCTION _CCCL_FORCEINLINE static
+      cudaError_t Dispatch(void *d_temp_storage, size_t &temp_storage_bytes, ..., cudaStream stream) {
+        if (/* nothing no items to process */) return cudaSuccess;
+
+        int ptx_version   = 0;
+        const cudaError_t error = CubDebug(PtxVersion(ptx_version));
+        if (cudaSuccess != error)
+        {
+          return error;
+        }
+        using MaxPolicy = typename SelectedPolicy::MaxPolicy;
+        DispatchAlgorithm dispatch(..., stream);
+        return CubDebug(MaxPolicy::Invoke(ptx_version, dispatch));
       }
-
     };
 
-For many algorithms, the dispatch layer is part of the API.
-The main reason for this integration is to support ``size_t``.
-Our API uses ``int`` as a type for ``num_items``.
-Users rely on the dispatch layer directly to workaround this.
-Exposing the dispatch layer also allows users to tune algorithms for their use cases.
-In the newly added functionality, the dispatch layer should not be exposed.
+While the dispatch layers of many algorithms are publicly accessible,
+and were used because they expose additional performance knobs,
+we no longer expose it publicly for new algorithms.
+
+For many algorithms, the dispatch layer is publicly accessible and used directly by users,
+since it often exposes additional performance knobs or configuration,
+like choosing the index type or policies to use.
+Exposing the dispatch layer also allowed users to tune algorithms for their use cases.
+In the newly added algorithms, the dispatch layer should not be exposed publicly anymore.
```

Comment:
question: this seem to be said in the previous paragraph already. Am I missing something?